crates/reliable-broadcast/src/lib.rs

```diff
@@ -94,12 +94,11 @@ where
                 fut.push(send_message(receiver, message.clone(), None));
             }
             while let Some((receiver, result)) = fut.next().await {
-                match result {
-                    Ok(msg) => {
-                        if let Ok(ack) = msg.try_into() {
-                            if let Ok(Some(aggregated)) = aggregating.add(receiver, ack) {
-                                return aggregated;
-                            }
+                match result.and_then(|msg| msg.try_into().map_err(|e| anyhow::anyhow!("{:?}", e)))
```

Comment:
we probably should log the error, potentially the aggregating should handle the error? IIUC this is because we wrap the rpc response with Result?

Reply:
Yes, because we return a Result<DAGMessage, DagRpcError>, this can be an Err(). But, try_into() from DAGNetworkMessage can fail as well in the malicious case (which we need to handle too).

I am not sure if we should log here in a library crate. Letting `aggregating` handle the error is good idea though. I am thinking we should retry for all Err() cases, otherwise we risk hitting the unreachable code.